features/FEATURE_BLE/targets/TARGET_CORDIO/source/CordioBLE.cpp

```diff
@@ -388,20 +392,18 @@ void BLE::callDispatcher()
     // process the external event queue
     _event_queue.process();
 
-    // follow by stack events
-    static uint32_t lastTimeUs = us_ticker_read();
-    uint32_t currTimeUs, deltaTimeMs;
+    int delta_ticks = (_timer.read_ms() / WSF_MS_PER_TICK);
 
-    // Update the current cordio time
-    currTimeUs = us_ticker_read();
-    deltaTimeMs = (currTimeUs - lastTimeUs) / 1000;
-    if (deltaTimeMs > 0) {
-        WsfTimerUpdate(deltaTimeMs / WSF_MS_PER_TICK);
-        lastTimeUs += deltaTimeMs * 1000;
+    if (delta_ticks > 0) {
+        WsfTimerUpdate(delta_ticks);
+
+        _timer.reset();
```

Comment:
this will stretch over time, will whatever this is running be compromised by this?